### DISCUSSION TITLE
adding work progress or status

### DISCUSSION THREAD

**Discussion by upsilon01** [PDMLab/docker-compose#188] [Ideas]
any ideas how to get the progress of a docker-compose command?
maybe to pipe the _stdout_ + _stderr_, and analyze it?

ty

**Comment by AlexZeitler** [comment]
@upsilon01 Do you mean more progress than `{ log: true }` option provides?

**Reply by upsilon01** [reply]
@AlexZeitler 
yes, `{ log: true } ` only outputs to the **calling process stdout**,
I want to analyze the output, and return the rate of progress in percentage

**Reply by AlexZeitler** [reply]
@upsilon01 Can you explain this please for a command you have in mind? What would be the progress of the `up` command over time?

**Reply by Steveb-p** [reply]
Basically what needs to happen is we need merge both external output stream and the one that we pass into the child process @AlexZeitler. This way both the promise can be resolved, and progress could be monitored from the calling code.

It's something that we wanted to cover by providing something akin to "Generator", but that would be a BC break.

Probably using https://nodejs.org/api/stream.html#streampipelinestreams-callback API, or one the others in that doc.

**Reply by upsilon01** [reply]
@AlexZeitler 
i would like to get a event\callback every time something happens, for example `up` command will have the following events:

`event: Starting test1_1 ...` 
`event: Starting postgres_1 ... `
`event: redis_1 is up-to-date`
`event: Starting test1_1    ... done`
`event: Starting postgres_1 ... done`

thank you.

**Reply by AlexZeitler** [reply]
@Steveb-p 

> It's something that we wanted to cover by providing something akin to "Generator", but that would be a BC break.

As we're still < 1.0.0 I think it should be ok.
